x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/utils.test.ts

```diff
@@ -21,3 +23,66 @@ describe('extractDomainAndEntryPointFromUrl', () => {
     });
   });
 });
+
+describe('getDomainWithProtocol', () => {
+  const { http } = mockHttpValues;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('passes through domain if it starts with https', async () => {
+    const result = await getDomainWithProtocol('https://elastic.co');
+
+    expect(result).toEqual('https://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('passes through domain if it starts with http', async () => {
+    const result = await getDomainWithProtocol('http://elastic.co');
+
+    expect(result).toEqual('http://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('returns domain with https protocol if back-end valdates https', async () => {
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: true }));
+    const result = await getDomainWithProtocol('elastic.co');
+
+    expect(result).toEqual('https://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(1);
+    expect(http.post).toHaveBeenCalledWith('/api/app_search/crawler/validate_url', {
+      body: JSON.stringify({ url: 'https://elastic.co', checks: ['tcp', 'url_request'] }),
+    });
+  });
+
+  it('returns domain with http protocol if back-end valdates http', async () => {
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: false }));
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: true }));
+    const result = await getDomainWithProtocol('elastic.co');
+
+    expect(result).toEqual('http://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(2);
+    expect(http.post).toHaveBeenLastCalledWith('/api/app_search/crawler/validate_url', {
+      body: JSON.stringify({ url: 'http://elastic.co', checks: ['tcp', 'url_request'] }),
+    });
+  });
+
+  it('passes through domain if back-end throws error', async () => {
+    http.post.mockReturnValueOnce(Promise.reject());
+
+    const result = await getDomainWithProtocol('elastic.co');
+
+    expect(result).toEqual('elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(1);
+  });
+
+  it('passes through domain if back-end fails to validate https and http', async () => {
```

Comment:
This is a cool distinction that I didn't catch on initial source code skim, thanks for detailing it so clearly 💯 Love this test file in general!

Reply:
Thanks Constance, I appreciate the recognition. I am beginning to appreciate the tests-as-documentation mindset.